dsl/interpreter.py

```diff
@@ -0,0 +1,164 @@
+import os
+import re
+import logging
+
+from dsl.parser import Grammar
+
+logger = logging.getLogger('example_logger')
+logging.basicConfig(format='%(asctime)s - %(message)s', level=logging.INFO)
+DSL_EXTENSION = '.str'
+MODULES_DIR_NAME = 'dsl/modules'
+modules = {}
+def load_modules():
+    content = os.listdir(MODULES_DIR_NAME)
+    for file in content:
+        if os.path.isfile(os.path.join(MODULES_DIR_NAME, file)) and file.endswith('.py'):
+            name = file[:len(file)-3]
+            modules[name] = exec()
+
+
+def parse_file(input_name: str) -> str | None:
+    if input_name.endswith('.py'):
+        return input_name
+    if not input_name.endswith(DSL_EXTENSION):
+        logger.info(f'File {input_name} has incorrect extension')
+        return None
+    if not os.path.isfile(input_name):
+        logger.error(f'File {input_name} does not exist')
+    file = open(input_name, "r")
+    lines = file.readlines()
+    # print whole file to a string with removing comments
+    is_commented = False
+    data = ''
+    # gr = Grammar()
+    # clean from comments
+    for line in lines:
+        line = line.strip()
+        ind_line_commented = len(line)
+        for i, c in enumerate(line):
+            if c == '@':
+                is_commented = not is_commented
+            if c == '#' and not is_commented and ind_line_commented == len(line):
+                ind_line_commented = i
+            if not (is_commented or ind_line_commented != len(line)):
+                data += c
+    # code = gr.parse(data)
+    code = simple_transform(data)
+    new_file_name = input_name[:len(input_name) - len(DSL_EXTENSION)] + '.py'
```

Comment:
Using pathlib with `input_name` of type `Path`
```suggestion
    new_file_name = input_name.with_suffix(".py")
```